youtube_dl/extractor/compoundmedia.py

```diff
@@ -0,0 +1,87 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+import re
+import sys
+
+from ..utils import (
+    ExtractorError,
+    urlencode_postdata,
+)
+
+from .livestream import LivestreamIE
+
+
+class CompoundMediaIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?compoundmedia\.com/shows/[^/]+/(?P<id>[0-9]+)'
+    _LOGIN_URL = 'https://www.compoundmedia.com/api/login'
+    _METADATA_URL_PREFIX = 'https://www.compoundmedia.com/api/video/'
+    _NETRC_MACHINE = 'compoundmedia'
+
+    # No tests because all videos require authentication
+    # _TEST = {
+    #     'url': 'https://www.compoundmedia.com/shows/tacs/1851',
+    #     'md5': '756a17ea88a671d9faafc6aa406abdb0',
+    #     'info_dict': {
+    #         'id': '1851',
+    #         'title': 'TACS 001 - The First Episode',
+    #     }
+    # }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        username, password = self._get_login_info()
+        if username is None:
+            self.raise_login_required()
+
+        data = {
+            'email': username,
+            'password': password,
+        }
+
+        login_page, url_handle = self._download_json_handle(
+            self._LOGIN_URL, None, 'Logging in',
+            data=urlencode_postdata(data), headers={
+                'Content-Type': 'application/x-www-form-urlencoded',
+                'Referer': self._LOGIN_URL,
+            })
+
+        if login_page.get('token') is None:
+            raise ExtractorError('Invalid username or password', expected=True)
+
+        for header, cookies in url_handle.headers.items():
+            if header.lower() != 'set-cookie':
+                continue
+            if sys.version_info[0] >= 3:
+                cookies = cookies.encode('iso-8859-1')
+            cookies = cookies.decode('utf-8')
+            cfduid = re.search(
+                r'__cfduid=(.+?);.*?\bdomain=(.+?)(?:[,;]|$)', cookies)
+            if cfduid:
+                value, domain = cfduid.groups()
+                self._set_cookie(domain, '__cfduid', value)
```

Comment:
What's the point of this? `set-cookie` headers are handled internally.

Reply:
@dstftw I was just copying from another. So all Set-Cookie's are persisted for future requests?

Reply:
Have you even bothered to read the rationale comment from where you copy pasted it?

Reply:
@dstftw No, sorry. I've removed the code and it works.